python/check_hex_string.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Daniel Estevez <daniel@destevez.net>
+#
+# This file is part of gr-satellites
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+
+from gnuradio import gr
+import pmt
+
+
+class check_hex_string(gr.basic_block):
+    """docstring for block check_hex_string"""
+    def __init__(self, hexstring, startindex=0):
+        gr.basic_block.__init__(
+            self,
+            name='check_hex_string',
+            in_sig=[],
+            out_sig=[])
+
+        self.hexstring = bytes.fromhex(hexstring)
+        self.startindex = startindex
+
+        self.message_port_register_in(pmt.intern('in'))
+        self.set_msg_handler(pmt.intern('in'), self.handle_msg)
+        self.message_port_register_out(pmt.intern('ok'))
+        self.message_port_register_out(pmt.intern('fail'))
+
+    def handle_msg(self, msg_pmt):
+        msg = pmt.cdr(msg_pmt)
+        if not pmt.is_u8vector(msg):
+            print('[ERROR] Received invalid message type. Expected u8vector')
+            return
+        packet = bytes(pmt.u8vector_elements(msg))
+
+        # Check packet length
+        if len(packet) < len(self.hexstring):
+            self.message_port_pub(pmt.intern('fail'), msg_pmt)
+            return
```

Comment:
This check should also take into account `self.startindex`.

Reply:
I've tested it with a startindex greater than the length of the string at input port. In this case it behaves as expected, the string from port "in" will be passed to port "fail". So, I would leave the startindex unchecked, if ok with you.

Reply:
The potential problem is that in the line
```python
hex_str = packet[self.startindex:self.startindex + len(self.hexstring)]
```
below, if `len(packet) < self.startindex + len(self.hexstring)`, then you're slicing outside of the `packet` length. In Python this works, but you'll get a `hex_str` of the wrong length (shorter, potentially empty). This won't pass the equality match against `self.hexstring`. So either you can remove the check in L40-L43 altogether because Python's slicing takes care of failing the match if the packet is too short, or correct the L40-L43 packet length check to account for `self.startindex`.

Reply:
decided to remove the check in L40-L43